app/extensions/safe/server-routes/safe.js

```diff
@@ -19,19 +25,60 @@ const safeRoute = {
                 return reply( 'SAFE not connected yet' );
             }
 
-            const data = await app.webFetch( link );
+            if ( headers.range )
+            {
+                const range = headers.range;
+                const rangeArray =
+                  range.substring( BYTES.length, range.length )
+                      .replace( /['"]+/g, '' )
+                      .split( '-' );
+
+                start = rangeArray[0] ? parseInt( rangeArray[0], 10 ) : null;
+                end = rangeArray[1] ? parseInt( rangeArray[1], 10 ) : null;
+
+                //we need to separate out 0- length requests which are not partial content
+                if( start && start !== 0 )
+                {
+                    //should have a start !== 0
+                    isRangeReq = true;
+                }
+
+                if( start === 0 && end )
+                {
+                    isRangeReq = true;
+                }
+            }
+
+            // setup opts object
+            const options = { headers };
+
+            if ( isRangeReq )
+            {
+                options.range = { start, end };
+            }
+
+            const data = await app.webFetch( link, options );
+
+            if ( headers.range )
+            {
+                return reply( data.body )
+                    .code( 206 )
+                    .type( data.headers['Content-Type'] )
+                    .header( 'Content-Range', data.headers['Content-Range'] )
+                    .header( 'Content-Length', data.headers['Content-Length'] );
+            }
 
             return reply( data.body )
                 .type( data.headers['Content-Type'] )
-                .header( 'Content-Encoding', 'chunked');
+                .header( 'Transfer-Encoding', 'chunked' );
         }
         catch ( e )
```

Comment:
I think we should also send a different code when there is a problem with the ranges, taken from https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests:
```
In case of a range request that is out of bounds (range values overlap the extent
of the resource), the server responds with a 416 Requested Range Not Satisfiable status.
```